### DISCUSSION TITLE
iOS Supabase SDK Upload Errors

### DISCUSSION THREAD

**Discussion by nickarner-meter** [supabase/supabase#18569] [Questions]
I have a table I setup in Supabase, partially shown here:

 
  
![CleanShot 2023-10-29 at 14 49 12@2x](https://github.com/supabase/supabase/assets/119962635/3ec2217f-ea3c-49b3-85ec-3fe321ec53f8)

I'm using the Supabase iOS SDK to send data to the table:

```
   
   class SupabaseAnalyticsManager: ObservableObject {
        static let shared = SupabaseAnalyticsManager()
        private var client: SupabaseClient?
        
        var lastSentClockDrawRecords: [PathData] = []
        var lastSentClockCopyRecords: [PathData] = []
        
        private init() {
            if let url = URL(string: "myURL") {
                client = SupabaseClient(supabaseURL: url, supabaseKey: "mykey")
            } else {
                print("Invalid Supabase URL")
            }
        }

    func insertPathDataForClockCopy(records: [PathData]) async throws {
        if records == lastSentClockCopyRecords {
            return
        }
        
        lastSentClockCopyRecords = records
        
        guard let client = client else {
            print("Error: Supabase client is not initialized.")
            throw SupabaseError.clientNotInitialized
        }

        do {
            let response = try await client.database.from("clock-copy").insert(values: records).execute()
            print("Batch Insert Response for Clock Copy Path Data: \(response)")
            
            if response.status != 201 {
                print("Error: Batch insertion of Clock Copy path data failed with status code \(response.status)")
                throw SupabaseError.insertionFailed
            }
        } catch {
            print("Error: An exception occurred while batch inserting the Clock Copy path data records. Error details: \(error)")
            throw error
        }
    }

    enum SupabaseError: Error {
        case clientNotInitialized
        case insertionFailed
    }
   ...
}

```

But I keep getting `Error: An exception occurred while batch inserting the Clock Copy path data records. Error details: unacceptableStatusCode(404) Failed to send data to Supabase: unacceptableStatusCode(404)`.

I know that my url and my API key are correct

**Comment by GaryAustin1** [comment]
404 on insert/update is usually you don't meet the RLS policy if you have RLS enabled.

**Reply by nickarner-meter** [reply]
Hmm; I currently *don't* have RLS enabled for any of my tables

**Reply by GaryAustin1** [reply]
You have it disabled or it is enabled and you don't have policies?
404 means the table can't be found or accessed.  Could be bad name, RLS, table grants have been changed by something like Prisma, self-hosted setup as an issue.  Can you do a select at same spot?

**Reply by nickarner-meter** [reply]
Oh I'm an idiot. I had done "-" instead of "_" in my client-side code. 
Ok, now that's resolved. Now I have to figure out why I'm getting back `Error: An exception occurred while batch inserting the Clock Draw path data records. Error details: unacceptableStatusCode(400)`

**Reply by GaryAustin1** [reply]
400 usually means your request is bad format.  I don't know much about swift, but this is just one hit I got of a formatting issue with the data being sent:  https://github.com/supabase-community/supabase-swift/issues/114
